src/main/java/com/hierynomus/smbj/share/Directory.java

```diff
@@ -38,8 +38,12 @@
     }
 
     public <F extends FileDirectoryQueryableInformation> List<F> list(Class<F> informationClass) throws SMBApiException {
+        return list(informationClass, null);
+    }
+
+    public <F extends FileDirectoryQueryableInformation> List<F> list(Class<F> informationClass, String searchPattern) {
```

Comment:
Maybe add a bit of JavaDoc to the method to define what can go into a `searchPattern`